src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package org.folio.consortia.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.consortia.domain.dto.UserTenant;
+import org.folio.consortia.domain.dto.UserTenantCollection;
+import org.folio.consortia.domain.entity.UserTenantEntity;
+import org.folio.consortia.domain.mapper.UserTenantMapper;
+import org.folio.consortia.domain.repository.UserTenantRepository;
+import org.folio.consortia.exception.ResourceNotFoundException;
+import org.folio.consortia.service.UserTenantService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class UserTenantServiceImpl implements UserTenantService {
+
+  private final UserTenantRepository userTenantRepository;
+  private final UserTenantMapper userTenantMapper;
+
+  @Transactional(readOnly = true)
+  @Override
+  public UserTenantCollection get(Integer offset, Integer limit) {
+    var result = new UserTenantCollection();
+    Page<UserTenantEntity> userTenantPage = userTenantRepository.findAll(PageRequest.of(offset, limit));
+    result.setUserTenants(userTenantPage.stream().map(userTenantMapper::toDto).toList());
+    result.setTotalRecords((int) userTenantPage.getTotalElements());
+    return result;
+  }
+
+  @Override
+  public UserTenant getById(UUID id) {
+    Optional<UserTenantEntity> userTenantEntity = userTenantRepository.findById(id);
+    if (userTenantEntity.isEmpty()) {
+      throw new ResourceNotFoundException("associationId", String.valueOf(id));
+    }
+    return userTenantMapper.toDto(userTenantEntity.get());
+  }
+
+  @Transactional(readOnly = true)
+  @Override
+  public UserTenantCollection getByUserId(UUID userId) {
+    var result = new UserTenantCollection();
+    UserTenantEntity userTenantEntity = userTenantRepository.findByUserId(userId)
+      .orElseThrow(() -> new ResourceNotFoundException("userId", String.valueOf(userId)));
+    result.setUserTenants(List.of(userTenantMapper.toDto(userTenantEntity)));
+    result.setTotalRecords(1);
+    return result;
+  }
+
+  @Transactional(readOnly = true)
```

Comment:
Lets remove this annoatation in order to make code simpler

Reply:
OK, I will remove